price_feed_generator/construct_yaml.py

```diff
@@ -0,0 +1,43 @@
+import csv
+import json
+import yaml
+from collections import defaultdict
+
+
+def load_coins(filepath: str) -> dict[str, list[dict]]:
+    with open(filepath, 'r') as file:
+        entries = json.loads(file.read())
+        res = defaultdict(list)
+        for entry in entries:
+            res[entry['symbol']].append(entry)
+        return res
+
+
+def load_tokens(filepath: str):
+    with open(filepath, newline='') as file:
+        return list(csv.DictReader(file))
+
+
+if __name__ == "__main__":
+    coins = load_coins('data/coinpaprikacoins.json')
+    tokens = load_tokens('data/tokens_without_prices.csv')
+
+    print("Total unlisted tokens", len(tokens))
+    found = 0
+    output = ""
+    for token in tokens:
+        if token['symbol'] in coins:
+            # print(token, coins[token['symbol']])
+            possible_coins = coins[token['symbol']]
+            if len(possible_coins) == 1:
+                coin = possible_coins[0]
+                if coin['is_active'] == True and coin['is_new'] == False:
+                    # For some reason dune uses the snake case id as the name
+                    coin_name = coin['id'].replace('-', '_')
+                    output += f"- name: {coin_name}\n  id: {coin['id']}\n  symbol: {token['symbol']}\n  address: {token['address']}\n  decimals: {token['decimals']}\n"""
```

Comment:
I'd prefer to have some library build the yaml instead of manually building it by concatenating strings. Yaml is weird so for example a token called `false` would be written as a yaml boolean and not as a string, similarly, a token called `"quoted"` would just be called `quoted`. I didn't see any tokens like this in the list however.

From [here](https://stackoverflow.com/questions/12470665/how-can-i-write-data-in-yaml-format-in-a-file/12471272#12471272) it looks like it should be possible to build yaml from dics without extra libraries.

Reply:
I had looked at importing the python yaml library but the specific syntax required didn't seem any easier than just building the string. I will continue looking into this.

Reply:
I has also stumbled upon the link you shared about yaml (I'm pretty sure this requires `import yaml`) which is part of the standard library. Anyway, even that example doesn't give me the impression of how I will git a hyphen on the `name` field, but none others. I will try it out anyway.

Reply:
I guess I could do this

```
d = {'- name': token_name, 'id': , ...}
```
like with the hyphen in the name, but it seems almost as hackey as the string concatenation.

Reply:
The hypen is usually used to represent vector entries in yaml, so for example this:
```
data = [
    dict(A = 'a'),
    [
     	'c',
	'd',
	'e',
    ],
    'test',
]
```
becomes this:
```
- A: a
- - c
  - d
  - e
- test
```

It's a pity that it requires an external library however.

Reply:
Quite nicely, if you used `data={'- name': 'token_name', 'id': 1}` then the ouput of the example code would be the correctly escaped version:
```
'- name': token_name
id: 1
```

Reply:
So I have pushed the code with my trial run of something that almost works. However the token addresses come with quotes and I can't seem to get rid of them.